x/subscription/keeper/subscription.go

```diff
@@ -172,47 +188,34 @@ func (k Keeper) CreateSubscription(
 			"plan":  planIndex,
 			"block": strconv.FormatInt(ctx.BlockHeight(), 10),
 		}
-		return utils.LavaError(ctx, logger, "subscribe", details, "invalid plan")
+		return utils.LavaError(ctx, logger, "CreateSubscription", details, "invalid plan")
 	}
 
-	// drop Plan reference in case of error:
-	// NOTE! FROM HERE ONWARD MUST SET ERR IF AN ERROR OCCURS
-	defer func() {
-		if err != nil {
-			k.plansKeeper.PutPlan(ctx, planIndex, plan.Block)
-		}
-	}()
-
 	err = k.projectsKeeper.CreateDefaultProject(ctx, consumer)
 	if err != nil {
 		details := map[string]string{
 			"err": err.Error(),
 		}
-		err = utils.LavaError(ctx, logger, "subscribe", details, "failed to create default project")
-		return err
+		return utils.LavaError(ctx, logger, "CreateSubscription", details, "failed to create default project")
 	}
 
-	// delete default Project in case of error
-	defer func() {
-		if err != nil {
-			// TODO: delete all projects of the subscription
-			// k.projectsKeeper.DeleteAllProject(ctx, consumer)
-			_ = err
-		}
-	}()
-
 	price := plan.GetPrice()
-	expiry := time.Now().UTC()
 
-	duration := int(plan.GetDuration())
-	for i := 0; i < duration; i++ {
-		expiry = nextMonth(expiry)
+	// use current block's timestamp for subscription start-time
+	timestamp := ctx.BlockTime()
+	expiry := timestamp
+
+	if duration < 12 {
+		price.Amount = price.Amount.MulRaw(int64(duration))
+		for i := 0; i < int(duration); i++ {
+			expiry = nextMonth(expiry)
+		}
 	}
 
-	if isYearly && duration < 12 {
+	if duration == 12 {
```

Comment:
can buy more than 12, and discount should apply, also can be an else case for the less than 12 if case prior

Reply:
FYI, it was decided to limit duration to 12 months per purchase, and also DurationLeft may not exceed 13 months (max duration + 1 for current month in which we renew before expiry). But I'll adjust to code in case that limit changes.